coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/PlaybackToSecondary.scala

```diff
@@ -216,34 +220,67 @@ class PlaybackToSecondary[CT, CV](u: PlaybackToSecondary.SuperUniverse[CT, CV],
       }
     }
 
+    sealed abstract class Consolidatable
+    object Consolidatable {
+      def all = Set[Consolidatable](Rows, Rollups)
+    }
+    case object Rows extends Consolidatable
+    case object Rollups extends Consolidatable
+
     // A series of events is consolidatable if it is an update which only
-    // changes row data, which is to say if it has the form:
+    // changes row data or rollups, which is to say if it has the form:
     //    RowsChangedPreview
     //    zero or one Truncated
     //    zero or more RowDataUpdated
     //    LastModifiedChanged
     // or
+    //    zero or more intermingled RollupCreatedOrUpdated and RollupDropped
     //    LastModifiedChanged
```

Comment:
ok so I walked through this (old) logic, and while the set above makes sense, the thing I don't get here is that rather than "a single LastModifiedChanged" the code seems to actually be saying "a single LastModifiedChanged followed by exactly one other LogEvent of any type" ?

probably missing something but just jotting down my confusion before I switch over to the new logic

Reply:
So no, it says "a single LMC followed by exactly nothing" - `it.head` on line 228 doesn't advance the iterator, so the subsequent `it.next()` just pops the LMC out in order to let us see if it's the end of the version.

Reply:
ok, I think that was my confusion-- I was thinking that for an iteratior that contains for example [LMC], if head == LMC, then hasNext would be false (because of nothing being after LMC)